os/arch/arm/src/common/up_checkspace.c

```diff
@@ -38,11 +38,23 @@ extern uint32_t _sdata;
 extern uint32_t _edata;
 extern uint32_t _sbss;
 extern uint32_t _ebss;
+extern uint32_t _ram_kernel_text_start;
+extern uint32_t _ram_kernel_text_end;
+
+bool is_kernel_text_space(void *addr)
+{
+	/* Check if the address lies in the kernel text section range */
+	if ((addr >= (void *)&_stext && addr <= (void *)&_etext) ||
```

Comment:
Is it possible that `addr` reaches `_etext`? If not, `addr < (void *)&_etext` is enough.

Reply:
In that case, how about modifying `up_hardfault` also? It didn't check the equal to `us_textend`.